java/src/main/java/org/hyperledger/fabric/client/Status.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.client;
+
+import org.hyperledger.fabric.protos.gateway.CommitStatusResponse;
+
+import static org.hyperledger.fabric.protos.peer.TransactionPackage.TxValidationCode;
+
+/**
+ * Information about a transaction that is committed to the ledger.
+ */
+public final class Status {
+    private final String transactionId;
+    private final long blockNumber;
+    private final TxValidationCode code;
+
+    Status(final String transactionId, final CommitStatusResponse response) {
+        this.transactionId = transactionId;
+        this.blockNumber = response.getBlockNumber();
+        this.code = response.getResult();
+    }
+
+    /**
+     * Get the transaction ID.
+     * @return A transaction ID.
+     */
+    public String getTransactionId() {
+        return transactionId;
+    }
+
+    /**
+     * Get the block number in which the transaction committed. If the transaction has not yet committed, this method
+     * blocks until the commit occurs.
```

Comment:
I guess it blocks on the getStatus() call now rather than here?

Reply:
Yes, you're right. That needs fixing